modules/gapi/src/streaming/onevpl/cfg_param_device_selector.hpp

```diff
@@ -0,0 +1,41 @@
+// This file is part of OpenCV project.
+// It is subject to the license terms in the LICENSE file found in the top-level directory
+// of this distribution and at http://opencv.org/license.html.
+//
+// Copyright (C) 2021 Intel Corporation
+
+#ifndef GAPI_STREAMING_ONEVPL_CFG_PARAM_DEVICE_SELECTOR_HPP
+#define GAPI_STREAMING_ONEVPL_CFG_PARAM_DEVICE_SELECTOR_HPP
+
+#include <opencv2/gapi/streaming/onevpl/device_selector_interface.hpp>
+#include <opencv2/gapi/streaming/onevpl/cfg_params.hpp>
+#include <opencv2/gapi/streaming/onevpl/source.hpp>
+
+#include "opencv2/gapi/own/exports.hpp" // GAPI_EXPORTS
+
+namespace cv {
+namespace gapi {
+namespace wip {
+namespace onevpl {
+
+struct GAPI_EXPORTS CfgParamDeviceSelector final: public IDeviceSelector {
+    CfgParamDeviceSelector(const CfgParams& params = {});
+    CfgParamDeviceSelector(Device::Ptr device_ptr,
+                           Context::Ptr ctx_ptr,
+                           const CfgParams& params);
```

Comment:
It'd be great to comment what's the difference between two - in the Doxygen format

Also, the first constructor might require `explicit`

Reply:
If we rejected adding IDeviceSelector into CfgParams for backend context creation, i would hide this selector implementation in private scope which just transparent meaning for usage in

```
oneVPLSource(...)  : 
   selector(new CfgParamDeviceSelector(...)) {}

oneVPLSource(Device_ptr, Ctx_ptr, Params) : 
    selector(new CfgParamDeviceSelector(Device_ptr, Ctx_ptr, params)) {}
```